### PR TITLE
Improve mobile support of views

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
This PR improves the responsiveness of the login, registration

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Use your browser dev tools to emulate a mobile device
- The login, registration & forgot password view should all look ok and should not overflow the screen width
- This also goes for the login view hidden in the navbar while viewing a public annotation / dataset and not being logged in.
  - Create an annotation, make it public, copy the sharing link.
  - Log out and open the sharing link on the mobile device browser tab
  - In the navbar hit the user icon -> the auth popup should look ok :)
- Open the dataset dashboard in the mobile browser tab. Compared to before, the dataset table should at least be a little visible and when working with a vertical mobile device, the table should now be kinda usable compared to before.

### TODOs:
- [ ] Any other ideas of views to make more responsive?

### Issues:
- No issue exists for this

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)

### CODE REVIEW

**Review by hotzenklotz — APPROVED** [review]
Works well. Great work.

**Comment by hotzenklotz** [comment]
The component `<CoverWithLogin />` could also benefit from the same treatment.

File: frontend/javascripts/components/brain_spinner.tsx [hunk 1]
```diff
@@ -106,7 +106,7 @@ export function CoverWithLogin({ onLoggedIn }: { onLoggedIn: () => void }) {
         }}
         align="middle"
       >
-        <Col span={8}>
+        <Col xs={22} sm={20} md={16} lg={12} xl={8}>
           <h3>Try logging in to view the dataset.</h3>
           <LoginForm layout="horizontal" onLoggedIn={onLoggedIn} />
```
Comment (MichaelBuessemeyer) [inline frontend/javascripts/components/brain_spinner.tsx:111]: But the `<CoverWithLogin/>` component is already adjusted right here

To me it looks like this: 
image
image

Comment (MichaelBuessemeyer) [reply frontend/javascripts/components/brain_spinner.tsx:111]: For record: The image shows the `<LoginView/>` component. To get the `<CoverWithLogin />` component rendered, try to open a non public dataset in view mode without being logged in

**Pull request merged**